Created commands

kristoferssolo opened the pull request:
Created 3 vim commands for user to run:
1. `:Runner` - wraps `require('runner').run()`;
2. `:AutoRunner` - wraps `require('runner').autorun()`;
3. `:AutoRunnerStop` - wraps `require('runner').autorun_stop()`;

The `AutoRuner` command automatically runs the code every time any file is saved.
The `AutoRunerStop` command stops `AutoRunner` and closes the window.

kristoferssolo committed:
Created `Runner` command

kristoferssolo committed:
Remove unnesessary temp variable

kristoferssolo committed:
Created `AutoRunner` command

`AutoRunner` command executes the `Runner` command every time a file is saved.

kristoferssolo committed:
Updated formatting

kristoferssolo committed:
Fix descriptions

kristoferssolo committed:
Created `AutoRunnerStop` command

`AutoRunnerStop` - stops `AutoRunner` and closes its window

kristoferssolo committed:
Automatic formatting

kristoferssolo committed:
Added command description to readme

MarcHamamji commented on README.md:
```diff
@@ -53,8 +54,20 @@ A customizable Neovim plugin to run code inside the editor
 
+### Commands
+
+| Command           | Description                              |
```
Add a table column to show which function each command runs, so it's more clear.

MarcHamamji commented on README.md:
```diff
@@ -53,8 +54,20 @@ A customizable Neovim plugin to run code inside the editor
+  require('runner').autorun_stop()
 
   -- Can also be called with the buffer number where the code is:
   -- require('runner').run( <buffer_number> )
```
Specify that both functions `run` and `autorun` can accept the buffer number as an argument.

MarcHamamji commented on lua/runner/init.lua:
```diff
@@ -28,29 +28,44 @@ M.set_handler = function(filetype, handler)
+    group = vim.api.nvim_create_augroup('AutoRunner', { clear = true }),
+    pattern = '*',
+    callback = function()
+      helpers.shell_handler(utils._last_command, false)()
```
You're restricting the command to only be a shell command. That way, if for example you're running a lua file, which uses the command handler to run `luafile %`, it breaks. Please update it to run the last handler regardless of which type it is.
You can add a `_last_handler` function somewhere that you overwrite each time you run the file, that just encapsulates the handler being run. When `autorun` runs, it just calls that function.

MarcHamamji reviewed (changes requested):
Thank you very much for your work! I added some comments for things I would like you to update. :)

kristoferssolo committed:
Update table

Added lua command column to "Commands" table.

kristoferssolo committed:
Update function description

Specified that `autorun` can accept buffer number as an argument.

kristoferssolo committed:
Fix `autorun` function

When saving a file, now function now checks if `_terminal_windows` exists and call the last shell command, otherwise it calls the vim command.

MarcHamamji commented:
Great, I just have one more question: Is there a reason you create the user commands in the `plugin` directory? Why not in the `setup()`?

kristoferssolo commented:
I followed examples of other plugins and [this template](https://github.com/m00qek/plugin-template.nvim). Everyone inserts code that needs to be executed when the plugin is loaded into the `plugins` directory, e.g. `user_commands`. Some even put `setup()` in there.

MarcHamamji commented:
Reading [this](https://www.reddit.com/r/neovim/comments/y58zh8/comment/isio0xk/?utm_source=share&utm_medium=web2x&context=3), I think that it's better to just put everything in the setup, so that the commands are only created when the `setup` function is called.

MarcHamamji reviewed (changes requested):
Reading [this](https://www.reddit.com/r/neovim/comments/y58zh8/comment/isio0xk/?utm_source=share&utm_medium=web2x&context=3), I think that it's better to just put everything in the setup, so that the commands are only created when the `setup` function is called. So can you please just copy the contents of that file and put them in the setup? Also can you modify the README to say that the setup is not optional anymore?

kristoferssolo committed:
Moved `user_commands`

`user_commands` now are created when `setup()` is called.

kristoferssolo committed:
Fix README

Removed the line stating that the `setup()` function in optional.

kristoferssolo commented:
Hopefully I understood you correctly

MarcHamamji commented:
Perfect, thanks a lot!
